projects/opencv/imdecode_fuzzer.cc

```diff
@@ -0,0 +1,15 @@
+#include <opencv2/opencv.hpp>
+#include <opencv2/imgcodecs/legacy/constants_c.h>
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  std::vector<uint8_t> image_data = {data, data + size};
+  cv::Mat data_matrix =
+      cv::Mat(1, image_data.size(), CV_8UC1, image_data.data());
```

Comment:
May be worth seeing if varying `CV_8UC1` improves coverage.

Reply:
I could add in support to randomly select a value from a collection of enum-values (e.g. {CV_8UC{1,2,3,4}, etc.}.

Might be easier to add this as a separate fuzz target? WDYT? This one is also nice since the corpus is the image data, if we use the input to select that matrix type things get a bit more complicated.

Reply:
You make a very good point about the corpus being images. Maybe just go with separate targets, or using a hash of the entire input to select the enum value.

Reply:
Do you mind if we leave this as a TODO? It wasn't too hard to make the change but I ran into some sanitizer errors with the change. I don't have the time at the moment to figure out if they are real or not (I'm guessing they are not real and due to some bug in the fuzz target).